Library/OcCryptoLib/BigNumLibInternal.h

```diff
@@ -0,0 +1,118 @@
+/**
+  Copyright (C) 2019, Download-Fritz. All rights reserved.
+
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#ifndef BIG_NUM_LIB_INTERNAL_H
+#define BIG_NUM_LIB_INTERNAL_H
+
+#include "BigNumLib.h"
+
+/**
+  Calculates the product of A and B.
+
+  @param[out] Hi  Buffer in which the high Word of the result is returned.
+  @param[in]  A   The multiplicant.
+  @param[in]  B   The multiplier.
+
+  @returns  The low Word of the result.
+
+**/
+OC_BN_WORD
+BigNumWordMul (
+  OUT OC_BN_WORD  *Hi,
+  IN  OC_BN_WORD  A,
+  IN  OC_BN_WORD  B
+  );
+
+/**
+  Calulates the difference of A and B.
+  A must have the same precision as B. Result must have a precision at most as
+  bit as A and B.
+
+  @param[in,out] Result  The buffer to return the result into.
+  @param[in]     A       The minuend.
+  @param[in]     B       The subtrahend.
+
+**/
+VOID
+BigNumDataSub (
+  IN OUT OC_BN_WORD        *Result,
+  IN     CONST OC_BN_WORD  *A,
```

Comment:
Could you please explain why `A` and `B` are pointers here, in BigNumDataPowMod, below, BigNumDataMontMulRow, etc?

Reply:
Because they are BigNums and not single words

Reply:
That's was unintuitive, yes… I confused the idea. Perhaps, we could really get rid of OC_BN and go with some packed format as I suggested above? This will make things simpler.

Reply:
Will mark as resolved as a duplicate of this: https://github.com/acidanthera/OcSupportPkg/pull/18#discussion_r341812754